shopinvader/services/cart.py

```diff
@@ -149,21 +150,77 @@ def _add_item(self, cart, params):
         else:
             with self.env.norecompute():
                 vals = self._prepare_cart_item(params, cart)
-                # the next statement is done with suspending the security for
-                #  performance reasons. It is safe only if both 3 following
-                # fields are filled on the sale order:
-                # - company_id
-                # - fiscal_position_id
-                # - pricelist_id
-                new_values = (
-                    self.env["sale.order.line"]
-                    .suspend_security()
-                    .play_onchanges(vals, vals.keys())
-                )
+                new_values = self._sale_order_line_onchange(vals)
                 vals.update(new_values)
-                self.env["sale.order.line"].create(vals)
+                self._create_sale_order_line(vals)
             cart.recompute()
 
+    @contextmanager
+    def _ensure_ctx_lang(self, values):
+        """
+        Todo: concurrent update still possible. We should find an improvement
+        (env.do_in_draft do SQL write so it's not a solution!)
+        Simulate the anonymous partner lang using the lang from the context.
+        To avoid to fill sale.order.line name/description with the anonymous
+        lang if the lang of the context is different.
+        This function update (in cache only) the anonymous partner's lang,
+        then you do your job (create etc) and the previous lang is
+        automatically reset with the original one.
+        Usage:
+        with self._simulate_anonymous_lang(vals):
+            # Do your job here
+        :param values: dict
+        :return:
+        """
+        order_id = values.get("order_id")
+        partner = self.env["sale.order"].browse(order_id).partner_id
+        anonymous_partner = self.shopinvader_backend.anonymous_partner_id
+        original_lang = partner.lang
+        ctx_lang = self.env.context.get("lang", partner.lang)
+        if (
+            partner
+            and anonymous_partner
+            and partner == anonymous_partner
+            and partner.lang != ctx_lang
+        ):
+            try:
+                partner.lang = ctx_lang
+                yield
+            finally:
+                partner.lang = original_lang
+        else:
+            yield
```

Comment:
@lmignon  I change the code (as the previous one was useless). Thank you again for your help! I hope it's fine this time!